lib/agent.js

```diff
@@ -214,9 +214,15 @@ Agent.prototype._handle = function handle(msg, rsinfo, outSocket) {
   }
 
   if (req.response) {
-    // it is an observe request
-    // and we are already streaming
-    return req.response.append(packet)
+    if (req.response.append) {
+      // it is an observe request
+      // and we are already streaming
+      return req.response.append(packet)
+    } else {
+      // TODO There is a previous response but is not an ObserveStream !
+      return
+    }
```

Comment:
I would prefer the request to be ACKed in this case, or am I missing
something?
Il giorno ven 24 lug 2015 alle 18:30 neich <notifications@github.com> ha
scritto:

> In lib/agent.js
> <https://github.com/mcollina/node-coap/pull/70#discussion_r35439414>:
>
> > @@ -214,9 +214,15 @@ Agent.prototype._handle = function handle(msg, rsinfo, outSocket) {
> >    }
> >
> >    if (req.response) {
> > -    // it is an observe request
> > -    // and we are already streaming
> > -    return req.response.append(packet)
> > +    if (req.response.append) {
> > +      // it is an observe request
> > +      // and we are already streaming
> > +      return req.response.append(packet)
> > +    } else {
> > +      // TODO There is a previous response but is not an ObserveStream !
> > +      return
> > +    }
>
> The spec says:
>
> https://tools.ietf.org/html/rfc7252#section-4.5
>
> 4.5.  Message Deduplication
>
>    A recipient might receive the same Confirmable message (as indicated
>    by the Message ID and source endpoint) multiple times within the
>    EXCHANGE_LIFETIME (Section 4.8.2), for example, when its
>    Acknowledgement went missing or didn't reach the original sender
>    before the first timeout.  The recipient SHOULD acknowledge each
>    duplicate copy of a Confirmable message using the same
>    Acknowledgement or Reset message but SHOULD process any request or
>    response in the message only once.  This rule MAY be relaxed in case
>    the Confirmable message transports a request that is idempotent (see
>    Section 5.1) or can be handled in an idempotent fashion
>
> We shoud process the message in the response only once, so I think it's ok
> to just return, right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mcollina/node-coap/pull/70/files#r35439414>.
>